[BUG] Ollama: explicitly built `ResponseFormat` with `JSON`/`TEXT` type not handled correctly

<!-- Please provide as many details as possible, this will help us to deliver a fix as soon as possible. Thank you! -->

**Describe the bug**
Ollama chat model throws `NullPointerException` when using explicitly built `TEXT` or `JSON` (without schema) response formats, due to identity comparison (`==`) in the conversion code between LC4j ResponseFormat and Ollama's response format.

**Log and Stack trace**
```
Caused by: java.lang.NullPointerException: Cannot invoke "dev.langchain4j.model.chat.request.json.JsonSchema.rootElement()" because the return value of "dev.langchain4j.model.chat.request.ResponseFormat.jsonSchema()" is null
	at dev.langchain4j.model.ollama.InternalOllamaHelper.toOllamaResponseFormat(InternalOllamaHelper.java:105)
	at dev.langchain4j.model.ollama.InternalOllamaHelper.toOllamaChatRequest(InternalOllamaHelper.java:181)
	at dev.langchain4j.model.ollama.OllamaChatModel.doChat(OllamaChatModel.java:39)
	at dev.langchain4j.model.chat.ChatModel.chat(ChatModel.java:46)
	at dev.langchain4j.model.chat.ChatModel.chat(ChatModel.java:76)
	at dev.langchain4j.model.ollama.OllamaStructuredOutputIT.lambda$should_accept_explicitly_built_text_response_format$0(OllamaStructuredOutputIT.java:240)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:71)
	... 6 more
```

**To Reproduce**
Provide explicitly built response formats when constructing `OllamaChatModel` instance. For example:
```
// This works
.responseFormat(ResponseFormat.JSON)

// This fails with NPE
.responseFormat(ResponseFormat.builder()
        .type(ResponseFormatType.JSON)
        .build())
```

Full example:
```
var model = OllamaChatModel.builder()
        .baseUrl(ollamaBaseUrl(ollama))
        .modelName(MODEL_NAME)
        .temperature(0.0)
        .responseFormat(ResponseFormat.builder()
                .type(ResponseFormatType.JSON)
                .build())
        .logRequests(true)
        .logResponses(true)
        .build();
```

**Expected behavior**
Ollama accepts explicitly built response formats and adheres to them.

**Please complete the following information:**
- LangChain4j version: 1.9.1
- LLM(s) used: `gemma3:1b`
- Java version: 17
- Spring Boot version (if applicable): 3.4.10

**Additional context**
<!-- Add any other context about the problem here. -->
None.

--- a/langchain4j-ollama/src/main/java/dev/langchain4j/model/ollama/InternalOllamaHelper.java
+++ b/langchain4j-ollama/src/main/java/dev/langchain4j/model/ollama/InternalOllamaHelper.java
@@ -9,14 +9,6 @@ import static dev.langchain4j.model.ollama.OllamaJsonUtils.fromJson;
 import static dev.langchain4j.model.ollama.OllamaJsonUtils.toJson;
 import static dev.langchain4j.model.ollama.OllamaJsonUtils.toJsonWithoutIdent;
 
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
-import java.util.Optional;
-import java.util.Set;
-import java.util.function.Predicate;
-import java.util.stream.Collectors;
 import com.fasterxml.jackson.core.type.TypeReference;
 import dev.langchain4j.Internal;
 import dev.langchain4j.agent.tool.ToolExecutionRequest;
@@ -36,10 +28,19 @@ import dev.langchain4j.exception.UnsupportedFeatureException;
 import dev.langchain4j.model.chat.request.ChatRequest;
 import dev.langchain4j.model.chat.request.ChatRequestParameters;
 import dev.langchain4j.model.chat.request.ResponseFormat;
+import dev.langchain4j.model.chat.request.ResponseFormatType;
 import dev.langchain4j.model.chat.request.json.JsonObjectSchema;
 import dev.langchain4j.model.chat.response.ChatResponseMetadata;
 import dev.langchain4j.model.output.FinishReason;
 import dev.langchain4j.model.output.TokenUsage;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
 @Internal
 class InternalOllamaHelper {
@@ -96,9 +97,9 @@ class InternalOllamaHelper {
     }
 
     static String toOllamaResponseFormat(ResponseFormat responseFormat) {
-        if (responseFormat == null || responseFormat == ResponseFormat.TEXT) {
+        if (responseFormat == null || responseFormat.type() == ResponseFormatType.TEXT) {
             return null;
-        } else if (responseFormat == ResponseFormat.JSON && responseFormat.jsonSchema() == null) {
+        } else if (responseFormat.type() == ResponseFormatType.JSON && responseFormat.jsonSchema() == null) {
             return "json";
         } else {
             return toJson(toMap(responseFormat.jsonSchema().rootElement()));
@@ -278,12 +279,12 @@ class InternalOllamaHelper {
                 .filter(Objects::nonNull)
                 .flatMap(List::stream)
                 .map(Content::type)
-                .collect(Collectors.toSet()) ;
+                .collect(Collectors.toSet());
         if (contentTypes.size() > 2) {
-            return false;   // contains more than 2 supported content types
+            return false; // contains more than 2 supported content types
         }
         contentTypes.remove(TEXT);
         contentTypes.remove(IMAGE);
-        return contentTypes.isEmpty();  // no remaining other content type
+        return contentTypes.isEmpty(); // no remaining other content type
     }
 }

--- a/langchain4j-ollama/src/test/java/dev/langchain4j/model/ollama/OllamaStructuredOutputIT.java
+++ b/langchain4j-ollama/src/test/java/dev/langchain4j/model/ollama/OllamaStructuredOutputIT.java
@@ -4,7 +4,12 @@ import static dev.langchain4j.model.ollama.AbstractOllamaLanguageModelInfrastruc
 import static dev.langchain4j.model.ollama.OllamaJsonUtils.fromJson;
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import dev.langchain4j.data.message.AiMessage;
 import dev.langchain4j.data.message.UserMessage;
 import dev.langchain4j.model.chat.ChatModel;
@@ -164,5 +169,82 @@ class OllamaStructuredOutputIT extends AbstractOllamaStructuredOutputLanguageMod
         assertThat(countryInfo.name()).isEqualTo("Canada");
         assertThat(countryInfo.capital()).isEqualTo("Ottawa");
         assertThat(countryInfo.languages()).contains("English", "French");
+    }
+
+    @Test
+    void should_accept_explicitly_built_json_response_format_without_schema() {
+        // given
+        var model = OllamaChatModel.builder()
+                .baseUrl(ollamaBaseUrl(ollama))
+                .modelName(MODEL_NAME)
+                .temperature(0.0)
+                .responseFormat(
+                        ResponseFormat.builder().type(ResponseFormatType.JSON).build())
+                .logRequests(true)
+                .logResponses(true)
+                .build();
+
+        // when
+        String prompt =
+                """
+                Tell me a joke.
+                Return ONLY a JSON object with fields "setup" and "punchline".
+                Do not use Markdown formatting.
+                Example: {"setup": "...", "punchline": "..."}
+                """;
+        String response = assertDoesNotThrow(
+                () -> model.chat(prompt),
+                "Model should accept explicitly built JSON (without schema) response format.");
+
+        // then
+        JsonNode responseJson;
+        try {
+            responseJson = new ObjectMapper().readTree(response);
+        } catch (JsonProcessingException e) {
+            fail(
+                    """
+                    Model response was expected to be a valid JSON but was not.
+                    Response content:"""
+                            + response,
+                    e);
+            return;
+        }
+
+        assertThat(responseJson.isObject()).isTrue();
+        assertThat(responseJson.size()).isEqualTo(2);
+        assertThat(responseJson.hasNonNull("setup")).isTrue();
+        assertThat(responseJson.hasNonNull("punchline")).isTrue();
+    }
+
+    @Test
+    void should_accept_explicitly_built_text_response_format() {
+        // given
+        var model = OllamaChatModel.builder()
+                .baseUrl(ollamaBaseUrl(ollama))
+                .modelName(MODEL_NAME)
+                .temperature(0.0)
+                .responseFormat(
+                        ResponseFormat.builder().type(ResponseFormatType.TEXT).build())
+                .logRequests(true)
+                .logResponses(true)
+                .build();
+
+        String secretCode = "8842";
+        String prompt =
+                """
+            Read this sentence: "The system override code is %s."
+            Extract the 4-digit code.
+            Return ONLY the code.
+            """
+                        .formatted(secretCode);
+
+        // when
+        String response = assertDoesNotThrow(
+                () -> model.chat(prompt), "Model should accept explicitly built text response format.");
+
+        // then
+        assertThat(response.trim())
+                .as("Model should extract the exact code provided in the prompt.")
+                .contains(secretCode);
     }
 }

